### ISSUE TITLE
Mouse clicks are displaced in fullscreen mode with fractional scaling

### ISSUE THREAD

**Issue by Aceler** [OpenRA/OpenRA#20863]
When running openra in fullscreen mode, the window still have a header bar, and that probably cause a mouse to click several pixels lower than the cursor position:

[2023-05-15 23-02-16.webm](https://github.com/OpenRA/OpenRA/assets/5177608/fd6d60bf-daaf-4420-8376-c9570d0547cd)

This only happens after I upgraded to 23.04. I tried flatpak, snap and appimage versions. 

```
$ /usr/bin/flatpak run --branch=stable --arch=x86_64 --command=openra-cnc --file-forwarding net.openra.OpenRA 
Platform is Linux (X64)
Engine version is release-20230225
Runtime: .NET CLR 6.0.14
Using SDL 2 with OpenGL (Modern) renderer
Desktop resolution: 2560x1440
No custom resolution provided, using desktop resolution
Using resolution: 2560x1440
Using window scale 1,00
OpenGL renderer: AMD Radeon RX 6600 (navi23, LLVM 15.0.7, DRM 3.49, 6.2.0-20-generic)
OpenGL version: 4.6 (Core Profile) Mesa 23.0.2 (git-4d5e73870e)
Using default sound device
Internal mods:
	ra: Red Alert (release-20230225)
	d2k: Dune 2000 (release-20230225)
	modcontent: Mod Content Manager (release-20230225)
	cnc: Tiberian Dawn (release-20230225)
External mods:
	cnc-release-20230225: Tiberian Dawn (release-20230225)
	d2k-release-20230225: Dune 2000 (release-20230225)
	ra-release-20230225: Red Alert (release-20230225)
Loading mod: cnc
```

I am using AMD RX6600, Gnome wayland session, fractional scale. I also tried to disable fractional scale, and disable scaling at all with no luck.

**Comment by HerrSnorre** [comment]
I have the same issue
Switching between x11 and wayland is useless
**Loging in (into a tty terminal) and running just `openra-ra` solved the issue**
Edit_01: **This may cause gnome to crash in another session**

"Graphics.DisableLegacyGL=False" did nothing

![image](https://github.com/OpenRA/OpenRA/assets/79027813/ae959168-80a9-42a9-bad8-7dee1937307e)
Note: the cursor offset its like 20px; the text on the buttons does not look good

```
[who@knows ~]$ openra-ra
Platform is Linux (X64)
Engine version is release-20230225
Runtime: .NET CLR 6.0.16
Using SDL 2 with OpenGL (Modern) renderer
Desktop resolution: 1920x1080
No custom resolution provided, using desktop resolution
Using resolution: 1920x1080
Using window scale 1,00
OpenGL renderer: NVIDIA GeForce GTX 1650/PCIe/SSE2
OpenGL version: 3.2.0 NVIDIA 530.41.03
Using default sound device
Internal mods:
	modcontent: Mod Content Manager (release-20230225)
	ra: Red Alert (release-20230225)
	cnc: Tiberian Dawn (release-20230225)
	d2k: Dune 2000 (release-20230225)
External mods:
	d2k-release-20210321: Dune 2000 (release-20210321)
	cnc-release-20230225: Tiberian Dawn (release-20230225)
	ra-release-20210321: Red Alert (release-20210321)
	d2k-release-20230225: Dune 2000 (release-20230225)
	ra-release-20230225: Red Alert (release-20230225)
Loading mod: ra
```
Another note: i did the same with the Intel UHD, and the same event ocorred

My setup (this is an hybrid laptop):
> Intel core i5 (UHD Graphics)
> Nvidia GTX 1650Ti (3.2.0 NVIDIA 530.41.03)

> OS: Arch Linux (kernel 6.3.4-arch1-1)
(and sorry for my bad inglesh)

**Comment by joopdo** [comment]
Same issue for me. Both Wayland and X11. Flatpak, Snap and appImage.

Running from terminal didn't solve it for me.
AMD + Radeon card.

````openra.ra 
Platform is Linux (X64)
Engine version is release-20230225
Runtime: .NET CLR 6.0.14
Using SDL 2 with OpenGL (Modern) renderer
Desktop resolution: 1920x1080
No custom resolution provided, using desktop resolution
Using resolution: 1920x1080
Using window scale 1.00
OpenGL renderer: AMD Radeon RX 570 Series (polaris10, LLVM 15.0.6, DRM 3.49, 6.2.0-20-generic)
OpenGL version: 4.6 (Core Profile) Mesa 22.2.5
Using default sound device
Internal mods:
	cnc: Tiberian Dawn (release-20230225)
	d2k: Dune 2000 (release-20230225)
	modcontent: Mod Content Manager (release-20230225)
	ra: Red Alert (release-20230225)
	ts: Tiberian Sun (release-20230225)
External mods:
	ra-release-20230225: Red Alert (release-20230225)
Loading mod: ra
````

**Comment by pchote** [comment]
This is caused by a regression in Gnome 44 / Ubuntu rather than OpenRA - see https://askubuntu.com/questions/1464849/title-bar-full-screen-in-23-04 and https://discourse.ubuntu.com/t/leading-up-to-23-04-release-things-to-consider/34879 for more details.

I have found a workaround which I will include in my planned Wayland improvements PR.

**Comment by Aceler** [comment]
Yeah, I don't see this bug in gnome 44 from other distros, tested in Fedora and ALT Linux. Thank you for your fixes!

**Comment by joopdo** [comment]
This issue still happens in i3wm (xorg) for me, switching to Gnome (wayland) solves it for me.

**Comment by Srandista** [comment]
Same for me, I tried it on Linux Mint 22.1, in default session I can't press any button for content installation. When I logged to the Wayland session, I was able to went through that phase, and then set game to window mode, in which I can use mouse  in X11 with no issues.

**Comment by Mailaender** [comment]
Does this still happen with https://github.com/OpenRA/OpenRA/releases/tag/playtest-20241228

**Comment by Srandista** [comment]
Yes, it does. I need to specify, that I don't have a issue with just few pixels, but much larger discrepancy. I also just determined, it's behave like this for me with fractional scaling, that I use on my laptop.

I have 14" laptop with 1440p resolution, and both 100% and 200% resolution scale is unusable for me, therefore I'm using 150%. And with this scale and fullscreen game, I basically can't click on any button. Once I'm in game and change to window mode, game again start to recognize the right position of mouse cursor and I can control the game normally.

**Comment by PunkPun** [comment]
Is this still an issue?

**Comment by Srandista** [comment]
I tried the latest playtest (from end of February) and in it, yes, the issue still persist.

**Comment by joopdo** [comment]
My issue is gone (OpenRA in Wayland)

**Comment by Srandista** [comment]
I must specify, on Wayland, the issue wasn't present already back in last year (as I mentioned in my previous post), I'm now talking about X11.

**Comment by chrissound** [comment]
Also have this issue with xmonad, xorg while running windowed mode and scaling set at 150%.
